File: code/205.go
/*
 * @Author: lmio
 * @Date: 2023-02-13 23:52:18
 * @LastEditTime: 2023-02-19 23:28:22
 * @FilePath: /leetcode/code/205.go
 * @Description:205.同构字符串
 */
package code

func IsIsomorphic(s string, t string) bool {
	sTt := map[byte]byte{}
	for i := range t {
		ch := t[i]
		val, ok := sTt[ch]
		if ok {
			if val != ch {
				return false
			}
		}else {
			sTt[ch] = s[i]
		}
	}
	return true
}

File: code/205.py
'''
Author: lmio 2091319361@qq.com
Date: 2023-07-05 21:52:15
LastEditors: lmio 2091319361@qq.com
Description: 205. 同构字符串
'''

def isIsomorphic(s: str, t: str) -> bool:
    sTt = {}
    tTs = {}
    for i in range(len(t)):
        ti = t[i]
        si = s[i]
        if si in sTt and ti in tTs:
            if sTt[si] != ti:
                return False
        elif si not in sTt and ti not in tTs:
            sTt[si] = ti
            tTs[ti] = si
        else:
            return False
    return True